engine/gamesys/src/gamesys/components/comp_collision_object.cpp

```diff
@@ -760,14 +867,18 @@ namespace dmGameSystem
     static void RayCastCallback(const dmPhysics::RayCastResponse& response, const dmPhysics::RayCastRequest& request, void* user_data)
     {
         dmhash_t message_id;
-        uintptr_t descriptor;
-        uint32_t data_size;
-        void* message_data;
-        dmPhysicsDDF::RayCastResponse responsDDF;
-        dmPhysicsDDF::RayCastMissed missedDDF;
+
+        dmGameObject::HInstance instance = (dmGameObject::HInstance)request.m_UserData;
+        dmMessage::URL receiver;
+        receiver.m_Socket = dmGameObject::GetMessageSocket(dmGameObject::GetCollection(instance));
+        receiver.m_Path = dmGameObject::GetIdentifier(instance);
+        dmGameObject::Result message_result = dmGameObject::RESULT_OK;
+        CollisionWorld* world = (CollisionWorld*)user_data;
+        uintptr_t lua_listener = world->m_LuaListener;
+        dmMessage::URL* resultReceiver = lua_listener == 0 ? &receiver : &world->m_LuaListenerReceiver;
         if (response.m_Hit)
         {
-            CollisionWorld* world = (CollisionWorld*)user_data;
+            dmPhysicsDDF::RayCastResponse responsDDF;
```

Comment:
We don't need a new `Event` here because It will be exactly the same as `RayCastResponse`. So ray casts use the same messages as before.

Reply:
Typo + code style: "response_ddf"
(old code, but since it was touched, let's fix it :) )